django_saml2_auth/views.py

```diff
@@ -166,7 +168,11 @@ def acs(r):
     is_new_user = False
 
     try:
-        target_user = user_model.objects.get(username=user_name)
+        if settings.SAML2_AUTH.get('TRIGGER', {}).get('FIND_USER', None):
+            target_user = import_string(settings.SAML2_AUTH['TRIGGER']['FIND_USER'])(user_identity)
+        else:
+            target_user = user_model.objects.get(username=user_name)
```

Comment:
This shouldn't probably be inside try-catch. IMHO the trigger shouldn't raise, but rather return None if no user is found.

Actually this whole section is so hard to read that it would be better to rewrite it.

Use variables for the `find_user_trigger_name` and its imported value:
```python
find_user_spec = settings.SAML2_AUTH.get('TRIGGER', {}).get('FIND_USER')
if find_user_spec:
    find_user = import_string(find_user_spec)
else:
    def find_user(username):
        return user_model.objects.filter(username=username).first()

target_user = find_user(user_name)
```

Reply:
I don't like that either, but we don't currently have time to rewrite.
Your suggestion does not work here because the trigger call must have user_identity as parameter instead user_name.

Reply:
So something like this then:
```
find_user_spec = settings.SAML2_AUTH.get('TRIGGER', {}).get('FIND_USER')
if find_user_spec:
    find_user = import_string(find_user_spec)
    target_user = find_user(user_identity)
else:
    target_user = user_model.objects.filter(username=username).first()

Reply:
That should be ok.